lighthouse-cli/performance-experiment/report/scripts/perf-x.js

```diff
@@ -0,0 +1,158 @@
+/**
+ * @license
+ * Copyright 2016 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* global window, document, location, fetch */
+
+'use strict';
+
+/**
+ * @fileoverview Report script for Project Performance Experiment.
+ *
+ * Include functions for supporting interation between report page and Perf-X server.
+ */
+
+window.addEventListener('DOMContentLoaded', () => {
+  new ConfigPanel();
+});
+
+class ConfigPanel {
+  constructor() {
+    this._configPanel = document.querySelector('.js-config-panel');
+    this._messageField = this._configPanel.querySelector('.js-message');
+    this._urlBlockingList = this._configPanel.querySelector('.js-url-blocking-patterns');
+    this._urlBlockingStatus = {};
+
+    const bodyToggle = this._configPanel.querySelector('.js-panel-toggle');
+    bodyToggle.addEventListener('click', this._toggleBody.bind(this));
```

Comment:
`bodyToggle.addEventListener('click', () => this._toggleBody);` should get you binding to `this` for free